src/Owin.Security.OpenIdConnect.Server/OpenIdConnectServerHandler.Endpoints.cs

```diff
@@ -1625,97 +1649,235 @@ internal partial class OpenIdConnectServerHandler : AuthenticationHandler<OpenId
+            // See https://tools.ietf.org/html/rfc7662#section-2.1
+            AuthenticationTicket ticket = null;
+            string hint;
+            if (request.Parameters.TryGetValue(OpenIdConnectConstants.Parameters.TokenTypeHint, out hint))
+            {
+                switch (hint)
+                {
+                    case OpenIdConnectConstants.TokenHintTypes.AccessToken:
+                        ticket = await ReceiveAccessTokenAsync(request.Token, request);
+                        break;
+                    case OpenIdConnectConstants.TokenHintTypes.RefreshToken:
+                        ticket = await ReceiveRefreshTokenAsync(request.Token, request);
+                        break;
+                    case OpenIdConnectConstants.TokenHintTypes.IdToken:
+                        ticket = await ReceiveIdentityTokenAsync(request.Token, request);
+                        break;
+                }
+            }
 
-                return;
+            // Note: if the coken can't be found using the hint or hint is not sent, search all supported token types.
+            // See https://tools.ietf.org/html/rfc7662#section-2.1
+            if (ticket == null)
+            {
+                ticket = await ReceiveAccessTokenAsync(request.Token, request);
+                if (ticket != null)
+                {
+                    hint = OpenIdConnectConstants.TokenHintTypes.AccessToken;
+                }
+                else
+                {
+                    ticket = await ReceiveIdentityTokenAsync(request.Token, request);
+                    if (ticket != null)
+                    {
+                        hint = OpenIdConnectConstants.TokenHintTypes.IdToken;
+                    }
+                    else
+                    {
+                        ticket = await ReceiveRefreshTokenAsync(request.Token, request);
+                        if (ticket != null)
+                        {
+                            hint = OpenIdConnectConstants.TokenHintTypes.RefreshToken;
+                        }
+                    }
+                }
             }
 
-            if (!ticket.Properties.ExpiresUtc.HasValue ||
-                 ticket.Properties.ExpiresUtc < Options.SystemClock.UtcNow) {
-                Options.Logger.WriteError("expired token");
+            ////TODO: Rework this code so it handles the token given in the parameter and NOT
+            ////  the token given for authentication.
+            var payload = new JObject();
+            if (ticket == null)
+            {
+                Options.Logger.WriteVerbose("invalid token");
+                //Handles cases when the request is valid but token is invalid or not deserialized into a ticket
+                payload.Add("active", false);
+            }
+            else
+            {
+                // TODO: Determine if this is still valid since requests to the endpoint "must" be authenticated/authorized
```

Comment:
Yes, I think it's still valid, but it will need a serious rework. Remove it for now and I'll take care of it in a future PR :+1: